docs/examples/DataSet/Deeper dive on qcodes doNd functionalities with a focus on arrayed sweeping benefits.ipynb

```diff
@@ -0,0 +1,455 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Deeper dive on qcodes doNd functionalities with a focus on arrayed sweeping benefits"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this example notebook, we want to show how the qcodes measurement can be simplified by using qcodes doNd functions. Also, we test 2-dimentional sweeping with a naive point by point sweeping over x and y axis and measure z axis values compared to sweeping over x axis and considering y axis as an array of numbers to measure z axis values. The array method can be useful in buffered data acquisitions from instruments that have buffer to write data in it (this notebook is not aiming to show how to setup buffer). Therefore, the measurement time can be enhanced significantly."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Necessary imports"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from qcodes import initialise_or_create_database_at\n",
+    "import time\n",
+    "import numpy as np\n",
+    "from qcodes.instrument.parameter import Parameter\n",
+    "from qcodes import load_or_create_experiment, Measurement,\\\n",
+    "load_by_run_spec\n",
+    "from qcodes.utils.dataset import doNd\n",
+    "from qcodes.dataset.plotting import plot_dataset"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Creating a database or loading it"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "initialise_or_create_database_at('test.db')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 1. Point by point 2D sweep over x and y"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The following cell is a naive way of generating an example 2D plot and time it. This uses single sweep points for both x and y directions, so it is quit slow."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Starting experimental run with id: 40. \n",
+      "Meaurement took 287.8640215396881 seconds\n"
+     ]
+    }
+   ],
+   "source": [
+    "xvals = np.linspace(-5, 5, 51)\n",
+    "yvals = np.linspace(-5, 5, 51)\n",
+    "xx, yy = np.meshgrid(xvals, yvals, indexing='ij')\n",
+    "zz = np.cos(xx) * np.sin(yy)\n",
+    "\n",
+    "def very_simple_2d_sweep():\n",
```

Comment:
i think that extracting the "loops" into helper functions makes the narrative more confusing than it helps. i think that this is a good case where a comment "very_simple_2d_sweep" next to the "loops" inside the measurement context manager will actually work better than the function. Moreover, perhaps these names/comments should be more specific? for example, point_by_point_x_y_sweep and point_by_point_x_sweep_with_y_buffered_sweep (along these lines but much better)?